components/horn-client/src/main.rs

```diff
@@ -0,0 +1,202 @@
+/*******************************************************************************
+* Copyright (c) 2024 Contributors to the Eclipse Foundation
+*
+* See the NOTICE file(s) distributed with this work for additional
+* information regarding copyright ownership.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0
+*
+* SPDX-License-Identifier: EPL-2.0
+*******************************************************************************/
+
+use std::path::PathBuf;
+use std::sync::Arc;
+use env_logger::Env;
+use log::error;
+use log::info;
+use up_rust::communication::{CallOptions, InMemoryRpcClient, RpcClient, UPayload};
+use up_transport_zenoh::UPTransportZenoh;
+use up_transport_zenoh::zenoh_config;
+use clap::Parser;
+
+use horn_proto::horn_service::{ ActivateHornRequest, ActivateHornResponse, DeactivateHornRequest};
+use horn_proto::horn_topics::{HornMode, HornSequence, HornCycle};
+ 
+ #[tokio::main]
+ async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    const ACTIVATE_RESOURCE_ID: u16 = 0x0001;
+    const DEACTIVATE_RESOURCE_ID: u16 = 0x0002;
+    env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
+ 
+    info!("Starting the client for the COVESA Horn service over uProtocol");
+
+    let transport = Arc::new(
+        UPTransportZenoh::new(get_zenoh_config(), "//horn_client/1/1/0")
+            .await
+            .unwrap(),
+    );
+     
+    // The Zenoh transport happens to implement the 
+    //traits for UTransport and LocalUriProvider, 
+    //which is why it is used twice here.
+    let rpc_client = InMemoryRpcClient::new(transport.clone(), transport.clone()).await?;
+
+    let horn_request = ActivateHornRequest {
+        mode : HornMode::HM_SEQUENCED.into(),
+        command : vec![HornSequence {
+            horn_cycles: vec![HornCycle {
+                on_time: 100,
+                off_time: 100,
+                ..Default::default()
+            },
+            HornCycle {
+                on_time: 200,
+                off_time: 300,
+                ..Default::default()
+            },
+            HornCycle {
+                on_time: 100,
+                off_time: 200,
+                ..Default::default()
+            }, HornCycle {
+                on_time: 10000,
+                off_time: 500,
+                ..Default::default()
+            }],
+            ..Default::default()
+        }],
+        ..Default::default()
+    };
+
+    
+    let payload = UPayload::try_from_protobuf(horn_request)?;
+    match rpc_client
+        .invoke_method(
+            up_rust::UUri::try_from_parts("kuksa_server", 1, 1, ACTIVATE_RESOURCE_ID).unwrap(),
```

Comment:
This URI needs to reflect the service's authority name, service ID and version (ID and version can be found in the service's proto file).
```suggestion
            up_rust::UUri::try_from_parts("horn-service-kuksa", 28, 1, ACTIVATE_RESOURCE_ID).unwrap(),
```

Reply:
Yes